test/new-e2e/runner/profile.go

```diff
@@ -0,0 +1,102 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package runner
+
+import (
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/runner/parameters"
+)
+
+type CloudProvider string
+
+const (
+	AWS       CloudProvider = "aws"
+	Azure     CloudProvider = "az"
+	GCP       CloudProvider = "gcp"
+	EnvPrefix               = "E2E_"
+
+	envSep = ","
+)
+
+var (
+	runProfile  Profile
+	initProfile sync.Once
+)
+
+type Profile interface {
+	// EnvironmentName returns the environment names for cloud providers
+	EnvironmentNames() string
```

Comment:
Because this is how multiple environment has been implemented in `test-infra`, see https://github.com/DataDog/test-infra-definitions/pull/53/files#diff-b4e6cc5bd3d64c22ac609a026038e265d79e25aca8a94c10389c3cb7ebc7fff5

As a reminder, we can only pass string parameters in the ConfigMap at the moment.